web/src/features/moreCast2/forecastDraft.ts

```diff
@@ -0,0 +1,70 @@
+import { DraftMorecast2Rows, MoreCast2ForecastRow, MoreCast2Row } from 'features/moreCast2/interfaces'
+import { getRowsMap, isForecastRow } from 'features/moreCast2/util'
+import { DateTime } from 'luxon'
+
+export class MorecastDraftForecast {
+  public readonly STORAGE_KEY = 'morecastForecastDraft'
+  private readonly localStorage: Storage
+
+  constructor(localStorage: Storage) {
+    this.localStorage = localStorage
+  }
+
+  public getStoredDraftForecasts = (): DraftMorecast2Rows => {
+    const storedDraftString = this.localStorage.getItem(this.STORAGE_KEY)
+    let storedDraft: DraftMorecast2Rows = { rows: [], lastEdited: undefined }
+
+    if (storedDraftString) {
+      storedDraft = JSON.parse(storedDraftString)
+    }
+
+    return storedDraft
+  }
+
+  private storeDraftForecasts = (forecastDraft: DraftMorecast2Rows) => {
+    this.localStorage.setItem(this.STORAGE_KEY, JSON.stringify(forecastDraft))
+  }
+
+  public clearDraftForecasts = () => {
+    this.localStorage.removeItem(this.STORAGE_KEY)
+  }
+
+  public updateStoredDraftForecasts = (rowsToStore: MoreCast2Row[]) => {
+    const storedForecastsToUpdate = this.getStoredDraftForecasts()
+    const storedRowsMap = getRowsMap(storedForecastsToUpdate.rows)
+
+    rowsToStore.forEach(row => {
+      storedRowsMap.set(row.id, row)
+    })
+    // we only need to store rows that are 'Forecast' rows
+    storedForecastsToUpdate.rows = Array.from(storedRowsMap.values()).filter(row => {
+      return isForecastRow(row)
+    })
+    storedForecastsToUpdate.lastEdited = Date.now()
```

Comment:
It's not specified in the task but we should use a timezone with the date. For now we can stick to PST until we implement smart date.

You can use a `luxon` `DateTime` and pass it to the `pstFormatter` in `date.ts` since we've used that with HFI Calculator.

Reply:
`pstFormatter` doesn't handle times, and it turns out the `fromObject` method used in it doesn't convert time zones, it interprets the time being passed to it to have the time zone specified in `fromObject`. So if I gave it a DateTime.now(), it would return the wrong time since we're currently in PDT
Just a note for the future smart dates

Reply:
Maybe that's already obvious in the name `pstFormatter`